hypothesis-python/src/hypothesis/internal/conjecture/data.py

```diff
@@ -153,6 +168,100 @@ def __repr__(self):
 MAX_DEPTH = 100
 
 
+def calc_examples(self):
+    """Build the list of examples from either a ``ConjectureResult``
+    or a ``ConjectureData`` object by interpreting the recorded
+    example boundaries and parsing them into a tree of ``Example``
+    objects, returning the resulting list.
+
+    This is needed because we want to calculate these lazily.
+    The ``Example`` tree is fairly memory hungry and mildly
+    expensive to compute and, especially during generation, we
+    will often not need it, so we only want to compute it on
+    demand."""
+    example_stack = []
+    examples = []
+    index = 0
+
```

Comment:
I've realised the logic in here isn't quite right in terms of how it handles empty examples. Will fix shortly, but don't bother reviewing it until I have.